### PR TITLE
Sync states

### PR DESCRIPTION
**Pull request by yzhurko**

### CODE REVIEW

**Comment by yzhurko** [comment]
https://github.com/BlockSettle/common/pull/2520

File: GUI/QtQuick/AddressFilterModel.cpp [hunk 1]
```diff
@@ -12,10 +12,31 @@
 #include "AddressListModel.h"
 #include <QtGlobal>
 
-AddressFilterModel::AddressFilterModel(QObject* parent)
-   : QSortFilterProxyModel(parent)
+AddressFilterModel::AddressFilterModel(std::shared_ptr<SettingsController> settings)
+   : QSortFilterProxyModel()
```
Comment (sergey-chernikov) [inline GUI/QtQuick/AddressFilterModel.cpp:16]: Shouldn't there be a QtQuickAdapter as a parent anyway?

File: GUI/QtQuick/SettingsController.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   bool hasParam(ApplicationSettings::Setting key) const;
+   const QVariant& getParam(ApplicationSettings::Setting key) const;
+   void setParam(ApplicationSettings::Setting key, const QVariant& value);
+
+   const SettingsCache& getCache() const;
+   void resetCache(const SettingsCache& cache);
+
+signals:
+   void changed(ApplicationSettings::Setting);
+   void reseted();
```
Comment (sergey-chernikov) [inline GUI/QtQuick/SettingsController.h]: "reseted" doesn't exist in English. Use "reset".

File: GUI/QtQuick/AddressFilterModel.cpp (hunk 1 above)
Comment (yzhurko) [reply GUI/QtQuick/AddressFilterModel.cpp:16]: No, it's not necessary in the case. I don't use raw pointers to use object deletion via parent.

**Comment by yzhurko** [comment]
https://github.com/BlockSettle/common/pull/2521

**Review by sergey-chernikov — APPROVED** [review]

**Pull request merged**